### DISCUSSION TITLE
Are support for hotplug cpu + mem are planned?

### DISCUSSION THREAD

**Discussion by udoL** [siderolabs/talos#3927] [Ideas]
Hi,
talos (v.0.10) don't support hotplugging cpu and mem (running on proxmox) yet.
Additional CPUs are not reconiced and memory  shows 1GB only…
Is there an support for hotplugging on the road map?

Best regards
Udo

**Comment by andrewrynhard** [comment]
Hello @udoL.  Do you know what it takes in the kernel configuration to support this?

**Reply by udoL** [reply]
Hi @andrewrynhard 
two loaded kernel modules:
acpiphp + pci_hotplug

udev-rule for auto-add cpus:
/lib/udev/rules.d/80-hotplug-cpu.rules
SUBSYSTEM=="cpu", ACTION=="add", TEST=="online", ATTR{online}=="0", ATTR{online}="1"

For memory (+ actual kernel) kernel parameter at boot:
memhp_default_state=online

**Reply by andrewrynhard** [reply]
Awesome, thanks.
